downloads/swan-lake-release-notes/swan-lake-2201.3.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,460 @@
+
+## Update Ballerina
+
+**If you are already using Ballerina 2201.0.0 (Swan Lake)**, run either of the commands below to directly update to 2201.3.0 using the [Ballerina Update Tool](/learn/cli-documentation/update-tool/).
+
+`bal dist update` (or `bal dist pull 2201.3.0`)
+
+**If you are using a version below 2201.0.0 (Swan Lake)**, run the commands below to update to 2201.3.0.
+
+1. Run `bal update` to get the latest version of the Update Tool.
+
+2. Run `bal dist update` ( or `bal dist pull 2201.3.0`) to update your Ballerina version to 2201.3.0.
+
+However, if you are using a version below 2201.0.0 (Swan Lake) and if you already ran `bal dist update` (or `bal dist pull 2201.3.0`) before `bal update`, see [Troubleshooting](/downloads/swan-lake-release-notes/swan-lake-2201.0.0#troubleshooting) to recover your installation.
+
+## Install Ballerina
+
+If you have not installed Ballerina, then download the [installers](/downloads/#swanlake) to install.
+
+## Migrate from Swan Lake Beta releases
+
+>**Info:** If you have been using Swan Lake Beta releases, delete the `Dependencies.toml` files in your Ballerina packages when migrating to Ballerina 2201.3.0 (Swan Lake). 
+
+A few backward-incompatible changes have been introduced during the Swan Lake Beta program, and thereby, some of your existing packages may not compile with Ballerina 2201.3.0 (Swan Lake). Therefore, you need to delete the `Dependencies.toml` file to force the dependency resolver to use the latest versions of your dependencies. 
+
+## Language updates
+
+### New features
+
+#### Added a new field to the `display` annotation
+
+A new field named `kind` has been introduced to the `display` annotation to indicate the kind of data. Allowed values are `text`, `password`, and `file`.
+
+```ballerina
+public type RefreshTokenGrantConfig record {|
+    @display {
+        iconPath: "Field.icon",
+        label: "clientSecret field",
+        kind: "password"
+    }
+    string clientSecret;
+|};
+```
+
+#### Added support for function pointers with defaultable parameters
+
+Function pointers are now allowed with default values for parameters. Any expression can be used as the default value of a function pointer parameter.
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    int num1 = 100;
+    int num2 = 50;
+    int num3 = 25;
+    function (int a = 0, int b = 0, int c = 0) returns int total = getSum;
+
+    io:println(total()); // Prints `0`.
+    io:println(total(num1, num2)); // Prints `150`.
+    io:println(total(num1, num2, num3)); // Prints `175`.
+}
+
+function getSum(int num1, int num2, int num3) returns int {
+    return num1 + num2 + num3;
+}
+```
+
+### Improvements
+
+#### More improvements on working with optional fields
+
+If there is an optional `T x?;` field in a record, the absence of `x` is represented by nil where `T` does not allow nil.
```

Comment:
Previously, there is no way to represent the absence of an optional field. Therefore, the following case does not work.
```suggestion
If there is an optional `T x?;` field in a record, the absence of `x` is represented by nil where `T` does not allow nil.
```

```ballerina
type Topt record {
   int x?;
   int y?;
};

function foo() {    
    Topt t = {x: 2};
    var { x, y } = t;
}
```
But with this new feature, the absence of an optional field can be represented by nil